packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -211,15 +211,20 @@ export default createRule<Options, MessageId>({
       }
       const property = node.property;
 
-      if (property.type === AST_NODE_TYPES.Identifier) {
```

Comment:
This condition is unnecessary because the property type is `Identifier | PrivateIdentifier`